jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -980,6 +979,143 @@ ecma_delete_property (ecma_object_t *object_p, /**< object */
+          {
+            new_length = index + 1;
+
+            if (new_length == old_length)
+            {
+              /* Early return. */
+              return new_length;
+            }
+          }
+        }
+      }
+    }
+
+    current_prop_p = ECMA_GET_POINTER (ecma_property_header_t,
+                                       current_prop_p->next_property_cp);
+  }
+
+  /* Second all properties between new_length and old_length are deleted. */
+  current_prop_p = ecma_get_property_list (object_p);
+  ecma_property_header_t *prev_prop_p = NULL;
+  bool has_hashmap = false;
+
+  if (ECMA_PROPERTY_GET_TYPE (current_prop_p->types[0]) == ECMA_PROPERTY_TYPE_HASHMAP)
+  {
+    prev_prop_p = current_prop_p;
+    current_prop_p = ECMA_GET_POINTER (ecma_property_header_t,
+                                       current_prop_p->next_property_cp);
+    has_hashmap = true;
+  }
+
+  while (current_prop_p != NULL)
+  {
+    ecma_property_pair_t *prop_pair_p = (ecma_property_pair_t *) current_prop_p;
+
+    for (int i = 0; i < ECMA_PROPERTY_PAIR_ITEM_COUNT; i++)
+    {
+      if (prop_pair_p->names_cp[i] != ECMA_NULL_POINTER
+          && ecma_is_property_configurable (current_prop_p->types[i]))
+      {
+        ecma_string_t *property_name_p = ECMA_GET_NON_NULL_POINTER (ecma_string_t,
+                                                                    prop_pair_p->names_cp[i]);
+
+        uint32_t index;
+        if (ecma_string_get_array_index (property_name_p, &index))
+        {
+          if (index < old_length && index >= new_length)
+          {
+            ecma_free_property (object_p, property_name_p, current_prop_p->types + i);
+
+            if (has_hashmap)
+            {
+              ecma_property_hashmap_delete (object_p, property_name_p, current_prop_p->types + i);
+            }
+
+            JERRY_ASSERT (current_prop_p->types[i] == ECMA_PROPERTY_TYPE_DELETED);
+
+            ecma_deref_ecma_string (property_name_p);
+            prop_pair_p->names_cp[i] = ECMA_NULL_POINTER;
+          }
+        }
+      }
+    }
+
+    if (current_prop_p->types[0] == ECMA_PROPERTY_TYPE_DELETED
+        && current_prop_p->types[1] == ECMA_PROPERTY_TYPE_DELETED)
+    {
+      if (prev_prop_p == NULL)
+      {
+        object_p->property_list_or_bound_object_cp = current_prop_p->next_property_cp;
+      }
+      else
+      {
+        prev_prop_p->next_property_cp = current_prop_p->next_property_cp;
+      }
+
+      ecma_property_header_t *next_prop_p = ECMA_GET_POINTER (ecma_property_header_t,
+                                                              current_prop_p->next_property_cp);
+      ecma_dealloc_property_pair ((ecma_property_pair_t *) current_prop_p);
+      current_prop_p = next_prop_p;
+    }
+    else
+    {
+      prev_prop_p = current_prop_p;
+      current_prop_p = ECMA_GET_POINTER (ecma_property_header_t,
+                                         current_prop_p->next_property_cp);
+    }
+  }
+
+  return new_length;
+}
```

Comment:
./jerry-core/ecma/base/ecma-helpers.c:1116: error: missing comment at the end of function: /* ecma_delete_array_properties */